config/remotes.go

```diff
@@ -0,0 +1,29 @@
+package config
+
+import (
+	"fmt"
+)
+
+// Remotes encapsulates configuration options for remotes
+type Remotes struct {
+	nameMap map[string]string
```

Comment:
because nameMap is unexported this field won't be visible to encoding packages like yaml & json, which will mean this data isn't written to disk (it's stored in yaml). We could refactor to just be:
```go
type Remotes map[string]string
```
and adjust `SetArbitrary` to alter `r *Remotes` values instead of `r.nameMap`.